### PR TITLE
Move UserEmailAddress to Shoebox

### PR DESCRIPTION
**Pull request by leogrim**
@eishay @andrewconner this also overrides EmailAddress.toString

As this removed primaryEmailId, all services (well at least Search and Eliza for sure) need to be deployed before Shoebox.

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/mail/ElizaUserEmailNotifierActor.scala [hunk 1]
```diff
@@ -110,22 +110,20 @@ class ElizaUserEmailNotifierActor @Inject() (
...
           log.warn(s"user $recipient is not active, not sending emails")
           Future.successful()
         } else {
-          val futureEmail = for {
-            destinationEmail <- shoebox.getEmailAddressById(recipient.primaryEmailId.get)
-            unsubUrl <- shoebox.getUnsubscribeUrlForEmail(destinationEmail)
-          } yield {
+          val destinationEmail = recipient.primaryEmail.get
+          val futureEmail = shoebox.getUnsubscribeUrlForEmail(destinationEmail).map { case unsubUrl =>
             val threadEmailInfo: ThreadEmailInfo = elizaEmailCommander.getThreadEmailInfo(thread, uriSummary, false, allUsers, allUserImageUrls, None, Some(unsubUrl), None, readTimeMinutesOpt).copy(pageUrl = deepUrl)
```
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/mail/ElizaUserEmailNotifierActor.scala:121]: @stkem @martinraison saving a trip to Shoebox here :)

Comment (martinraison) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/mail/ElizaUserEmailNotifierActor.scala:121]: nice!

**Comment by stkem** [comment]
@leogrim lgtm, however, given how pervasive this is, I would recommend just leaving it open until friday morning when you are back.

**Comment by leogrim** [comment]
All right, I'll work from a new branch!

**Comment by leogrim** [comment]
@stkem Since I won't be in the office on Friday, I'll have to merge this to get emails into the graph. I'll wait for you guys to be in the office Thursday morning. I'll double check the deployment order.

**Comment by stkem** [comment]
:+1:

**Pull request merged**